install.js

```diff
@@ -34,16 +34,16 @@ if (revisionInfo.downloaded)
   return;
 
 // Override current environment proxy settings with npm configuration, if any.
-const NPM_HTTPS_PROXY = process.env.npm_config_https_proxy || process.env.npm_config_proxy;
-const NPM_HTTP_PROXY = process.env.npm_config_http_proxy || process.env.npm_config_proxy;
-const NPM_NO_PROXY = process.env.npm_config_no_proxy;
+const HTTPS_PROXY = process.env.npm_config_https_proxy || process.env.npm_config_proxy || process.env.HTTPS_PROXY;
+const HTTP_PROXY = process.env.npm_config_http_proxy || process.env.npm_config_proxy || process.env.HTTP_PROXY;
+const NO_PROXY = process.env.npm_config_no_proxy || process.env.no_proxy || process.env.NO_PROXY;
 
-if (NPM_HTTPS_PROXY)
-  process.env.HTTPS_PROXY = NPM_HTTPS_PROXY;
-if (NPM_HTTP_PROXY)
-  process.env.HTTP_PROXY = NPM_HTTP_PROXY;
-if (NPM_NO_PROXY)
-  process.env.NO_PROXY = NPM_NO_PROXY;
+if (HTTPS_PROXY)
+  process.env.HTTPS_PROXY = HTTPS_PROXY;
```

Comment:
I'm not sure I understand how this patch changes the behavior. If the end goal is to have the process.env.HTTP_PROXY initialized with a global env variable, then it should already work on tip-of-tree.